test/functional/wallet_multiwallet.py

```diff
@@ -170,5 +170,24 @@ def run_test(self):
         assert_equal(w1.getwalletinfo()['paytxfee'], 0)
         assert_equal(w2.getwalletinfo()['paytxfee'], 4.0)
 
+        # dynamic wallet loading
+        self.restart_node(0, ['-nowallet'])
+        for wallet_name in wallet_names:
+            self.nodes[0].loadwallet(wallet_name)
```

Comment:
If I add a line here to inspect the wallet after the load:
```
self.nodes[0].getwalletinfo()
``` 
OR
```
self.nodes[0].getbalance()
```
I get the [json exception throw here](https://github.com/bitcoin/bitcoin/blob/master/src/wallet/rpcwallet.cpp#L78-L79) after the second load operation. Is this expected behavior?

Reply:
Yes - if there are multiple wallets loaded, you need to send the RPC request to the correct wallet endpoint. Look at `TestNode`'s `get_wallet_rpc()` method.

Reply:
Yes, good idea. I'll update the test